CHANGELOG.md

```diff
@@ -45,6 +45,7 @@
 * [BUGFIX] Compactor: Fix bug when using `-compactor.partial-block-deletion-delay`: compactor didn't correctly check for modification time of all block files. #2559
 * [BUGFIX] Query-frontend: fix wrong query sharding results for queries with boolean result like `1 < bool 0`. #2558
 * [BUGFIX] Fixed error messages related to per-instance limits incorrectly reporting they can be set on a per-tenant basis. #2610
+* [ENHANCEMENT] Querier: Set unlimited max concurrency in PromQL engine. #2598
```

Comment:
I would rephrase the CHANGELOG entry as "Querier: ensure all queries pulled from querier from query-frontend or query-scheduler are immediately executed. The maximum workers concurrency in each querier is configured by `-querier.max-concurrent`".


_Also, be aware we're used to keep CHANGELOG entries sorted by type. See [guidelines here](https://github.com/grafana/mimir/tree/main/docs/internal/contributing#changelog)._

Reply:
Changed.